# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1394733817 0
# Node ID 28b7881feb269ba7646decad69cc0c0e0ef4d87d
# Parent  66ebecf8c3e181ed8b9c30aabbf63500831f5987
tools/mfndump: Avoid unintentional NULL dereference

If we failed to open an xc interface, using xch to log an error will end in
tears.  Print to stderr instead, as we are bailing immediately later.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Coverity-id: 1191885
Acked-by: Ian Campbell <Ian.Campbell@citrix.com>
CC: Ian Jackson <Ian.Jackson@eu.citrix.com>
Acked-by: Dario Faggioli <dario.faggioli@citrix.com>

diff -r 66ebecf8c3e1 -r 28b7881feb26 tools/misc/xen-mfndump.c
--- a/tools/misc/xen-mfndump.c
+++ b/tools/misc/xen-mfndump.c
@@ -400,7 +400,7 @@ int main(int argc, char *argv[])
     xch = xc_interface_open(0, 0, 0);
     if ( !xch )
     {
-        ERROR("Failed to open an xc handler");
+        fprintf(stderr, "Failed to open an xc handler");
         return 1;
     }
 
